terraform/deployments/cluster-services/argo_workflows.tf

```diff
@@ -0,0 +1,46 @@
+locals {
+  tag_image_service_account_name = "add-tag-to-image-workflow"
+}
+
+module "tag_image_iam_role" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 5.5"
+
+  role_name        = "${local.tag_image_service_account_name}-${data.terraform_remote_state.cluster_infrastructure.outputs.cluster_id}"
+  role_description = "Role for the add-tag-to-image Argo Workflow. Corresponds to ${local.tag_image_service_account_name} k8s ServiceAccount."
+
+  oidc_providers = {
+    main = {
+      provider_arn               = data.terraform_remote_state.cluster_infrastructure.outputs.cluster_oidc_provider_arn
+      namespace_service_accounts = ["${var.apps_namespace}:${local.tag_image_service_account_name}"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "tag_image" {
+  statement {
+    actions = [
+      "ecr:GetAuthorizationToken",
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "ecr:BatchGetImage",
+      "ecr:PutImage",
+    ]
+    resources = ["arn:aws:ecr:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:repository/*"]
```

Comment:
This assumes that the registry is in-region, but I think that's a reasonable assumption (and moot for the time being anyway!) 👍